packages/core/src/providers/LocalMulticallProvider.tsx

```diff
@@ -0,0 +1,82 @@
+import { ReactNode, useEffect, useState } from 'react'
+import { isLocalChain } from '../helpers'
+import { useEthers } from '../hooks'
+import { useBlockNumber } from './blockNumber'
+import { useConfig, useUpdateConfig } from './config'
+import multicallABI from '../constants/abi/MultiCall.json'
+import { deployContract } from '../helpers/contract'
+
+interface LocalMulticallProps {
+  children: ReactNode
+}
+
+enum LocalMulticallState {
+  Unknown,
+  NonLocal,
+  Deploying,
+  Deployed,
+  Error,
+}
+
+export function LocalMulticallProvider({ children }: LocalMulticallProps) {
+  const updateConfig = useUpdateConfig()
+  const { multicallAddresses } = useConfig()
+  const { library, chainId } = useEthers()
+  const [localMulticallState, setLocalMulticallState] = useState(LocalMulticallState.Unknown)
+  const [multicallBlockNumber, setMulticallBlockNumber] = useState<number>()
+  const blockNumber = useBlockNumber()
+
+  useEffect(() => {
+    if (!library || !chainId) {
+      setLocalMulticallState(LocalMulticallState.Unknown)
+      return
+    }
+
+    if (!isLocalChain(chainId)) {
+      setLocalMulticallState(LocalMulticallState.NonLocal)
+      return
+    }
+
+    if (multicallAddresses && multicallAddresses[chainId]) {
+      setLocalMulticallState(LocalMulticallState.Deployed)
+      return
+    }
+
+    if (localMulticallState === LocalMulticallState.Deploying) {
+      return
+    }
+
+    const signer = library.getSigner()
+    if (!signer) {
+      setLocalMulticallState(LocalMulticallState.Error)
+      return
+    }
```

Comment:
I think it would be nicer style to have `if ... else if` rather than `return` statements, i.e.
```ts
    if (!library || !chainId) {
      setLocalMulticallState(LocalMulticallState.Unknown)
    } else if (!isLocalChain(chainId)) {
      setLocalMulticallState(LocalMulticallState.NonLocal)
    } else if (multicallAddresses && multicallAddresses[chainId]) {
      setLocalMulticallState(LocalMulticallState.Deployed)
    } else
    ...
```